File: Backend/Domain/Products/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Products
{
    public class Product
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public string BarCode { get; set; }

        public int Code { get; set;}

        public int AmountDaysPreviousNotification { get; set; }

        public bool IsActive { get; set; }

        public DateTime CreationDate { get; set; }

        public DateTime EndDate { get; set; }

        public Product()
        {

        }
    }
}

File: Frontend/vencimientos/src/app/models/products/product.ts
import { ProductSupplier } from "../suppliers/productSupplier";

export class Product {
  id!: number;
  name!: string;
  description!: string;
  barCode!: string;
  code!: number;
  amountDaysPreviousNotification!: number;
  isActive!: boolean;
  creationDate!: Date;
  endDate!: Date;

  supplier!: ProductSupplier;
  supplierId!: number;
  supplierName!: string;
}
